controllers/disruption_controller.go

```diff
@@ -58,6 +60,38 @@ type DisruptionReconciler struct {
 	InjectorDNSDisruptionKubeDNS          string
 	InjectorNetworkDisruptionAllowedHosts []string
 	ExpiredDisruptionGCDelay              time.Duration
+	Caches                                map[types.NamespacedName]context.CancelFunc
+}
+
+type SelectorPodsHandler struct {
+	disruption *v1beta1.Disruption
+}
```

Comment:
I know the PR is in progress still, but I don't see yet the benefit of this design, why use a struct that contains just a disruption to hold these methods?

Reply:
This PR is long outdated but it's there for example purposes only;
The goal is to make a handler fitting the handler interface with the 3 notifying methods; since there's going to be a unique handler per selector cache, and one dedicated selector cache per ongoing disruption, this handler will be triggered iif the corresponding disruption's selector cache is triggered
Having the disruption in the struct allows to have all the needed metadata to easily identify which selector was triggered